compiler/template/starlark/render.go

```diff
@@ -30,19 +31,20 @@ var (
 )
 
 // Render combines the template with the step in the yaml pipeline.
-func Render(tmpl string, name string, tName string, environment raw.StringSliceMap, variables map[string]interface{}, limit uint64) (*types.Build, error) {
+func Render(tmpl string, name string, tName string, environment raw.StringSliceMap, variables map[string]interface{}, limit int64) (*types.Build, error) {
 	config := new(types.Build)
 
 	thread := &starlark.Thread{Name: name}
-	// arbitrarily limiting the steps of the thread to 5000 to help prevent infinite loops
-	// may need to further investigate spawning a separate POSIX process if user input is problematic
-	// see https://github.com/google/starlark-go/issues/160#issuecomment-466794230 for further details
-	thread.SetMaxExecutionSteps(limit)
 
-	predeclared := starlark.StringDict{"struct": starlark.NewBuiltin("struct", starlarkstruct.Make)}
+	if limit < 0 {
+		return nil, fmt.Errorf("starlark exec limit must be non-negative")
+	}
 
-	globals, err := starlark.ExecFile(thread, tName, tmpl, predeclared)
+	thread.SetMaxExecutionSteps(uint64(limit))
+
+	predeclared := starlark.StringDict{"struct": starlark.NewBuiltin("struct", starlarkstruct.Make)}
 
+	globals, err := starlark.ExecFileOptions(syntax.LegacyFileOptions(), thread, "templated-base", tmpl, predeclared)
```

Comment:
does this change impact functionality?
```go
globals, err := starlark.ExecFileOptions(syntax.LegacyFileOptions(), thread, "templated-base", tmpl, predeclared)
```

Reply:
https://github.com/google/starlark-go/blob/1207426daebd7ea8eb86b8d917a344a72cc5167b/starlark/eval.go#L333

It's just using what the now deprecated function has. I think we can investigate turning some of those options on, but they're all set to false right now